server/src/main/java/io/deephaven/server/barrage/BarrageStreamGenerator.java

```diff
@@ -285,15 +283,18 @@ public SubView(final BarrageStreamGenerator generator,
             numModBatches = (numModRows + batchSize - 1) / batchSize;
 
             // precompute add row offsets
-            if (keyspaceViewport != null) {
-                try (WritableRowSet intersect = keyspaceViewport.intersect(generator.rowsIncluded.original)) {
-                    addRowOffsets = generator.rowsIncluded.original.invert(intersect);
-                }
-            } else if (!generator.rowsAdded.original.equals(generator.rowsIncluded.original)) {
-                // there are scoped rows included in the chunks that need to be removed
-                addRowOffsets = generator.rowsIncluded.original.invert(generator.rowsAdded.original);
+            if (generator.isSnapshot) {
+                // use chunk data as-is, rely on snapshot process to avoid duplicates
```

Comment:
I don't follow this change. Snapshots are sent to more than one user. We shouldn't send them data that they did not request. The `myAddedKeys` change below also seems suspicious. We should chat more about this change.

Reply:
You're correct this is unnecessary and the original code needs to be reverted.